hetmech/hetmat.py

```diff
@@ -39,6 +40,32 @@ def hetmat_from_graph(graph, path):
     return hetmat
 
 
+def hetmat_from_permuted_graph(hetmat, permutation_id, permuted_graph):
+    """
+    Assumes subdirectory structure and that permutations inherit nodes but not
+    edges.
+    """
+    if '.' in permutation_id:
+        raise ValueError('Permutations ids cannot contain `.`')
+    if not hetmat.permutations_directory.is_dir():
+        hetmat.permutations_directory.mkdir()
+    directory = hetmat.permutations_directory.joinpath(f'{permutation_id}.hetmat')
+    if directory.is_dir():
+        # If directory exists, back it up using a .bak extension
+        backup_directory = directory.with_name(directory.name + '.bak')
```

Comment:
Why backup existing directory by default?

Reply:
I was thinking that there could be substantial computing power cached in these directories and I didn't want someone accidentally overwritting everything.